client/shipping-label/state/actions.js

```diff
@@ -192,8 +197,13 @@ export const openPrintingFlow = () => (
 	dispatch( { type: OPEN_PRINTING_FLOW } );
 };
 
-export const exitPrintingFlow = ( force ) => {
-	return { type: EXIT_PRINTING_FLOW, force };
+export const exitPrintingFlow = ( force ) => ( dispatch, getState ) => {
+	const form = getState().shippingLabel.form;
+	if ( form.needsPrintConfirmation ) {
+		dispatch( clearAvailableRates() );
+	}
```

Comment:
Let's do this after dispatching the exit flow action so that we're:
1) consistent with the other calls
2) avoid the user seeing the rates being cleared (even if only for a second)